lib/Adapter/WorseReflection/Refactor/WorseGenerateMethod.php

```diff
@@ -57,10 +58,7 @@ public function generateMethod(SourceCode $sourceCode, int $offset, ?string $met
         $prototype = $this->addMethodCallToBuilder($methodCall, $visibility, $methodCall->isStatic(), $methodName);
         $sourceCode = $this->resolveSourceCode($sourceCode, $methodCall, $visibility);
 
-        return SourceCode::fromStringAndPath(
-            (string) $this->updater->textEditsFor($prototype, Code::fromString((string) $sourceCode))->apply($sourceCode),
-            $sourceCode->path()
```

Comment:
we need to add something like `TextDocumentEdits` value object to the text-document package and return that, and `WorkspaceEdits`

Reply:
Ahahaha, just noticed the same thing. Posted a discussion [here](https://github.com/phpactor/phpactor/discussions/1252).